src/examples/yaml_writer_demo.rb

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env ruby
+
+$LOAD_PATH.unshift(File.expand_path('../../lib', __FILE__))
+
+require "storage/fake_probing.rb"
+require "storage/fake_device_factory.rb"
+require "storage/yaml_writer.rb"
+
+input_file  = ARGV[0] || "fake-devicegraphs.yml"
+output_file = ARGV[1] || "/dev/stdout"
+
+fake_probing = Yast::Storage::FakeProbing.new
+devicegraph = fake_probing.devicegraph
+factory = Yast::Storage::FakeDeviceFactory.new(devicegraph)
+factory.load_yaml_file(input_file)
+
+yaml_writer = Yast::Storage::YamlWriter.new
+yaml_writer.write(devicegraph, output_file)
+
+
```

Comment:
I hate to start another discussion about APIs, but I have the feeling that we are going in the direction of having a `object_used_only_once = Whatever.new` every couple of lines. Something I fear can reveal wrong usage of object orientation and the single responsibility principle. And actually, this script is an perfect example of that because it contains, literally, a `Whatever.new` every couple of lines.

I believe that something like this feel way more natural for the average ruby developer (I'm not saying is perfect, though):
```ruby
factory = DevicegraphFactory.new
devicegraph = factory.create_from_file(input_file)
File.open(output_file, 'w') {|f| f.write devicegraph.to_yaml }
```
I did some prove of concept during the discussion of the previous pull request, and something like that can work without root permissions with no need to involve FakeProbing.